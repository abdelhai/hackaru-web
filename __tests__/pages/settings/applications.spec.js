import { Store } from 'vuex-mock-store';
import Factory from '@/__tests__/__setups__/factory';
import Applications from '@/pages/settings/applications';

describe('Applications', () => {
  let factory;
  let wrapper;

  const $store = new Store({
    getters: {
      'applications/getApplications': [
        {
          id: 1,
          name: 'Example1',
          uid: 'uid',
          redirectUri: 'urn:ietf:wg:oauth:2.0:oob',
          scopes: ['projects:read', 'projects:write'],
          confidential: true,
          createdAt: '2019-01-22T07:06:53.159Z',
          updatedAt: '2019-01-22T07:06:53.159Z',
          ownerId: null,
          ownerType: null
        },
        {
          id: 2,
          name: 'Example2',
          uid: 'uid',
          redirectUri: 'urn:ietf:wg:oauth:2.0:oob',
          scopes: ['activities:read', 'activities:write'],
          confidential: true,
          createdAt: '2019-01-22T07:06:53.159Z',
          updatedAt: '2019-01-22T07:06:53.159Z',
          ownerId: null,
          ownerType: null
        }
      ]
    }
  });

  beforeEach(() => {
    $store.reset();
    factory = new Factory(Applications, {
      mocks: { $store }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  it('dispatch applications/getApplications', () => {
    factory.shallow();
    expect($store.dispatch).toHaveBeenCalledWith(
      'applications/getApplications'
    );
  });

  describe('when click delete button and confirm is true', () => {
    beforeEach(() => {
      global.confirm = () => true;
      wrapper = factory.shallow();
      wrapper
        .findAll('.delete-button')
        .at(1)
        .vm.$emit('click');
    });

    it('dispatch applications/deleteApplication', () => {
      expect($store.dispatch).toHaveBeenCalledWith(
        'applications/deleteApplication',
        2
      );
    });

    it('send ga event', () => {
      expect(factory.options.mocks.$ga.event).toHaveBeenCalledWith(
        'oauth',
        'deleteApplication'
      );
    });
  });

  describe('when click delete button and confirm is false', () => {
    beforeEach(() => {
      global.confirm = () => false;
      wrapper = factory.shallow();
      wrapper
        .findAll('.delete-button')
        .at(1)
        .vm.$emit('click');
    });

    it('does not dispatch applications/deleteApplication', () => {
      expect($store.dispatch).not.toHaveBeenCalledWith(
        'applications/deleteApplication',
        2
      );
    });
  });

  describe('when click list item', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper
        .findAll('.list-item > h1')
        .at(1)
        .trigger('click');
    });

    it('render modal correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('show modal', () => {
      expect(factory.options.mocks.$modal.show).toHaveBeenCalledWith(
        'application'
      );
    });
  });
});
