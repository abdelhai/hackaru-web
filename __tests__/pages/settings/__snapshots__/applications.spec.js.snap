// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Applications render correctly 1`] = `
<div
  class="settings"
>
  <modal-wrapper-stub
    height="auto"
    name="application"
  >
    <form>
      <modal-header-stub>
        <h1>
          
        </h1>
      </modal-header-stub>
       
      <modal-item-stub
        class="is-vertical"
      >
        <modal-label-stub
          class="is-vertical"
        >
          
          labels.scopes
        
        </modal-label-stub>
         
        <highlight-stub>
          <ul />
        </highlight-stub>
      </modal-item-stub>
    </form>
  </modal-wrapper-stub>
   
  <section
    class="content"
  >
    <header
      class="header"
    >
      <heading-stub
        class="is-small"
      >
        <icon-stub
          name="box-icon"
        />
        
        title
      
      </heading-stub>
    </header>
     
    <div
      class="list-item"
    >
      <h1>
        Example1
      </h1>
       
      <btn-stub
        aria-label="ariaLabels.delete"
        class="delete-button has-icon"
      >
        <icon-stub
          class="is-danger"
          name="x-icon"
        />
      </btn-stub>
    </div>
    <div
      class="list-item"
    >
      <h1>
        Example2
      </h1>
       
      <btn-stub
        aria-label="ariaLabels.delete"
        class="delete-button has-icon"
      >
        <icon-stub
          class="is-danger"
          name="x-icon"
        />
      </btn-stub>
    </div>
     
    <!---->
  </section>
</div>
`;

exports[`Applications when click list item render modal correctly 1`] = `
<div
  class="settings"
>
  <modal-wrapper-stub
    height="auto"
    name="application"
  >
    <form>
      <modal-header-stub>
        <h1>
          Example2
        </h1>
      </modal-header-stub>
       
      <modal-item-stub
        class="is-vertical"
      >
        <modal-label-stub
          class="is-vertical"
        >
          
          labels.scopes
        
        </modal-label-stub>
         
        <highlight-stub>
          <ul>
            <li>
              
              scopes.activities:read
            
            </li>
            <li>
              
              scopes.activities:write
            
            </li>
          </ul>
        </highlight-stub>
      </modal-item-stub>
    </form>
  </modal-wrapper-stub>
   
  <section
    class="content"
  >
    <header
      class="header"
    >
      <heading-stub
        class="is-small"
      >
        <icon-stub
          name="box-icon"
        />
        
        title
      
      </heading-stub>
    </header>
     
    <div
      class="list-item"
    >
      <h1>
        Example1
      </h1>
       
      <btn-stub
        aria-label="ariaLabels.delete"
        class="delete-button has-icon"
      >
        <icon-stub
          class="is-danger"
          name="x-icon"
        />
      </btn-stub>
    </div>
    <div
      class="list-item"
    >
      <h1>
        Example2
      </h1>
       
      <btn-stub
        aria-label="ariaLabels.delete"
        class="delete-button has-icon"
      >
        <icon-stub
          class="is-danger"
          name="x-icon"
        />
      </btn-stub>
    </div>
     
    <!---->
  </section>
</div>
`;
